InnerContent, Modernize Underlying Libraries and Update Dependencies

kuasha420 opened the pull request:
Hello,

Great work with the library!

I was trying to add a custom component which required the innerContent to be passed to the block. So I had to make some changes to the library. 

While i was there, I also made react, react-dom and @loadable/component Peer Dependency,  Moved all dependencies to Dev Dependency and updated them to the latest version. This will ensure lesser package size and less chance of conflicts. 

I have also changed all instances of SFC to FC as the latter is depreciated. 

Please have a look and let me know if anything needs changing. 

Cheers!

kuasha420 committed:
Add innerContent, Update dependancies and make loadable/component peerDependancy

Also replaced deprecated SFC with FC and bumped minor version.
Update dist folder.

kuasha420 committed:
Update Readme to Mention TypeScript Peer Dependency

u12206050 commented on dist/blocks/common/audio.js:
```diff
@@ -4,7 +4,7 @@ const WPGAudioBlock = (props) => {
     // attrs,
     // innerBlocks,
-    innerHTML } = props;
+    innerHTML, } = props;
```
Remove comma

u12206050 commented on dist/blocks/common/cover.js:
```diff
@@ -4,7 +4,7 @@ const WPGCoverBlock = (props) => {
     // attrs,
     // innerBlocks,
-    innerHTML } = props;
+    innerHTML, } = props;
```
Remove comma

u12206050 commented on dist/blocks/common/file.js:
```diff
@@ -4,7 +4,7 @@ const WPGFileBlock = (props) => {
     // attrs,
     // innerBlocks,
-    innerHTML } = props;
+    innerHTML, } = props;
```
Remove comma

u12206050 commented on README.md:
````diff
@@ -98,25 +105,23 @@ import WPGBlocks from 'react-gutenberg'
-import WPGBlocks from "react-gutenberg"
-import { IWPGBlock } from "react-gutenberg/src/types"
+```tsx
+import * as React from "react";
````
Remove ;

u12206050 reviewed (changes requested):
Thanks for the effort. Could you remove the `;` at the end of the lines as it isn't really required in JS. Also remove the extra `,` commas that have been added. I commented on some of the lines as example.
Other than those two grammer issues the rest looks very good. Thanks.

kuasha420 commented:
@u12206050 oopsie boopsies. Looks like my prettier did it. Should I add a prettierrc file as well so these doesn't happen again? 

Cheers.

u12206050 commented:
Good idea.

On Wed, 16 Dec 2020, 15:29 Arafat Zahan, <notifications@github.com> wrote:

> @u12206050 <https://github.com/u12206050> oopsie boopsies. Looks like my
> prettier did it. Should I add a prettierrc file as well so these doesn't
> happen again?
>
> Cheers.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/u12206050/react-gutenberg/pull/6#issuecomment-746378263>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABSVMWROKXYXAGFLFWAQ7NTSVC73VANCNFSM4U52QQLA>
> .
>

kuasha420 committed:
Comply with projects grammer rules.

Install and configure prettier

Setup dist folder auto build
 - usable from from git ie. yarn add u12206050/react-gutenberg
 - no need to upload dist folder to git anymore
 - add dist/ to .gitignore
 - delete dist/ folder

kuasha420 commented:
@u12206050 I've pushed new commit. Please review. Thanks.

u12206050 commented on src/blocks/format/html.tsx:
```diff
@@ -2,12 +2,11 @@ import { IWPGBlock } from '../../types'
-    document.querySelectorAll(`[data-script="${this.id}"]`).forEach(script => {
-      (window as any).eval(script.innerHTML)
+    document.querySelectorAll(`[data-script="${this.id}"]`).forEach((script) => {
+      ;(window as any).eval(script.innerHTML)
```
uhm. Whats going on here?

kuasha420 replied:
Confused as well. Prettier keeps putting that semicolon back. Also do you prefer always or avoid for [arrow function parenthesis](https://prettier.io/docs/en/options.html#arrow-function-parentheses)?

u12206050 replied:
it can be avoid on function parenthesis.

kuasha420 replied:
Okay I'll configure it to avoid on my next PR.

u12206050 commented on src/blocks/index.tsx:
```diff
@@ -8,28 +8,44 @@ const DefaultHtmlBlock = loadable(() => import('./format/html'))
 
 export function GetTheBlock(name: string) {
   switch (name) {
-    case 'core/audio': return CommonBlocks.Audio
```
I think I much rather prefer the single one line instead of two lines. Makes it easier on the eyes to scan and read

kuasha420 replied:
Same, but I think it's [not possible](https://github.com/prettier/prettier/issues/6123) with prettier. One workaround is to add this file to prettierignore, but then we lose the benefit of prettier in this file. I'll do as you wish. 

Thanks.

u12206050 replied:
Ok no probem. Leave as is. I'll merge now but would be nice if the semicolon issue could be fixed

kuasha420 replied:
I'll try to fix that in my next pr where I fixed a warning about missing keys on column layouts. Thanks for the merge. Cheers!

u12206050 reviewed (changes requested):
Nearly there
